src/chips/add_carry_v1.rs

```diff
@@ -0,0 +1,162 @@
+use halo2_proofs::{circuit::*, halo2curves::pasta::Fp, plonk::*, poly::Rotation};
+
+#[derive(Debug, Clone)]
+pub struct AddCarryConfig {
+    pub advice: [Column<Advice>; 3],
+    pub instance: Column<Instance>,
+    pub selector: Selector,
+}
+
+#[derive(Debug, Clone)]
+pub struct AddCarryChip {
+    config: AddCarryConfig,
+}
+
+impl AddCarryChip {
+    pub fn construct(config: AddCarryConfig) -> Self {
+        Self { config }
+    }
+
+    pub fn configure(
+        meta: &mut ConstraintSystem<Fp>,
+        advice: [Column<Advice>; 3],
+        selector: Selector,
+        instance: Column<Instance>,
+    ) -> AddCarryConfig {
+        let col_a = advice[0];
+        let col_b = advice[1];
+        let col_c = advice[2];
+        let add_carry_selector = selector;
+
+        // Enable equality on the advice and instance column to enable permutation check
+        meta.enable_equality(col_a);
```

Comment:
The `enable_equality` on a column should be activated only if a copy constraint is present on a cell within that column. For example in this case there's no copy constraint performed on `col_a` so the activating the `enable_equality` can be safely removed

Reply:
Thanks for pointing this out!
Fixed at [fix: remove unnecessary enable_equality](https://github.com/summa-dev/halo2-experiments/pull/5/commits/eba3446d9108051a93e3c6506c62424a771a9b2a)